### ISSUE TITLE
Modrinth doesn't launch on unstable channel

### ISSUE THREAD

**Issue by JonasFranke** [NixOS/nixpkgs#359820]
## Describe the bug

The modrinth-app package does not launch when being on unstable channel

## Steps To Reproduce

1. add modrinth-app (or modrinth-app-unwrapped) to nixpkgs (unstable)
2. run nixos-rebuild
3. try to start modrinth

## Expected behavior

Modrinth starts as expected

## Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 24.05 (Uakari), 24.05.6818.6f6076c37180`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"home-manager-24.05.tar.gz, nixos-24.05, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

## Notify maintainers

@getchoo 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by JonasFranke** [comment]
> Do you have a NVIDIA card? There are a _lot_ of known bugs with webkitgtk and the drivers currently, including just segfaulting on launch

yep, is there a workaround? By e.g. using the integrated graphics?

**Comment by getchoo** [comment]
Integrated graphics is one solution (as Mesa shouldn't be affected by this), but you may also be able to set `WEBKIT_DISABLE_DMABUF_RENDERER=1` and/or `WEBKIT_DISABLE_COMPOSITING_MODE=1` to fix it. See https://github.com/tauri-apps/tauri/issues/9394 and https://github.com/tauri-apps/tauri/issues/9304 as well

**Comment by youwen5** [comment]
```
$ ModrinthApp
fish: Job 1, 'ModrinthApp' terminated by signal SIGSEGV (Address boundary error)
```

I get the above whenever I try to launch Modrinth. None of the environment variables seem to do anything. Any idea if this could be nixpkgs related, or should I open an issue in Modrinth?

**Comment by getchoo** [comment]
> Any idea if this could be nixpkgs related, or should I open an issue in Modrinth?

This is more likely to be a webkitgtk & NVIDIA issue rather than either us or Modrinth. It happens across all Tauri apps

**Comment by youwen5** [comment]
> > Any idea if this could be nixpkgs related, or should I open an issue in Modrinth?
> 
> 
> 
> This is more likely to be a webkitgtk & NVIDIA issue rather than either us or Modrinth. It happens across all Tauri apps

Weird, cause I do Tauri development on my local machine and I'm able to compile and run using both a Nix derivation and Cargo directly. I will investigate webkitgtk further, I guess.

**Comment by getchoo** [comment]
You'll probably want to see the upstream (Tauri) issues above. They in turn link to most of the webkitgtk bug reports

**Comment by CrazyWolf13** [comment]
Hi 
I can confirm this behaviour, I get this error:

```
2025-01-10T21:56:15.449730028+01:00  INFO theseus_gui: Initialized tracing subscriber. Loading Modrinth App!
2025-01-10T21:56:15.453688323+01:00  INFO theseus_gui: Initializing app...
```
Issue happens since I switched from default nixos to the proprietary nvidia drivers.

I've set:
```
	environment.sessionVariables = {
		"WEBKIT_DISABLE_DMABUF_RENDERER" = "1";
		"WEBKIT_DISABLE_COMPOSITING_MODE" = "1";
	};
```
And I'm also on wayland, though this did not fix anything, anything else I coud try?

**Comment by CrazyWolf13** [comment]
I finally found a fix on an older reddit thread:

adding `GDK_BACKEND=x11` makes it work again. 
(NixOS Wayland Gnome)

**Comment by CrazyWolf13** [comment]
@getchoo  
Is there any good way on adding this to my nixos config? Or even directly to the packagebuild?

Applying GDK_Backend=x11 globally to my whole system is not really something that seems like a good idea to me.

**Comment by getchoo** [comment]
Not tested, but I believe something like this should work

```nix
{ pkgs, ... }:

{
  environment.systemPackages = [
    (pkgs.modrinth-app.overrideAttrs (oldAttrs: {
      buildCommand =
        ''
          gappsWrapperArgs+=(
             --set GDK_BACKEND x11
             --set OTHER_ENVVAR VALUE
          )
        ''
        + oldAttrs.buildCommand;
    }))
  ];
}
```

(with `OTHER_ENVVAR` and `VALUE` being placeholders for anything else you may want to set)

**Comment by CrazyWolf13** [comment]
Thank you soooo much!
This snippet worked for me:
```
	environment.systemPackages =  
	with pkgs; [
		(modrinth-app.overrideAttrs (oldAttrs: {
			buildCommand = 
				''
					gappsWrapperArgs+=(
						--set GDK_BACKEND x11
						--set WEBKIT_DISABLE_DMABUF_RENDERER 1
					)
				''
				+ oldAttrs.buildCommand;
		}))

	];
```
